modules/EmbedPlayer/resources/mw.EmbedPlayer.js

```diff
@@ -3417,6 +3417,14 @@
 			//adaptive bitrate
 			return this.currentBitrate;
 		},
+		// get the target buffer from player, supporting only hls.js at this phase. 		
+		getTargetBuffer: function(){
+			var targetBuffer = null;
+			if(this.plugins.hlsjs){
```

Comment:
See how we implement this in hlsjs, again, the core shouldn't know about the plugin, it should be the other way around.